### PR TITLE
Replace crypto exchange apis with CCXT

### PR DESCRIPTION
**Pull request by jose-donato**
# Description

- [ ] Summary of the change / bug fix.
- [ ] Link # issue, if applicable.
- [ ] Screenshot of the feature or the bug before/after fix, if applicable.
- [ ] Relevant motivation and context. 
- [ ] List any dependencies that are required for this change.

# How has this been tested?

* Please describe the tests that you ran to verify your changes. 
* Provide instructions so we can reproduce. 
* Please also list any relevant details for your test configuration.

# Checklist:

- [ ] Update [our Hugo documentation](https://openbb-finance.github.io/OpenBBTerminal/) following [these guidelines](https://github.com/OpenBB-finance/OpenBBTerminal/tree/main/website).
- [ ] Update our tests following [these guidelines](https://github.com/OpenBB-finance/OpenBBTerminal/tree/main/tests).
- [ ] Make sure you are following our [CONTRIBUTING guidelines](https://github.com/OpenBB-finance/OpenBBTerminal/blob/main/CONTRIBUTING.md).
- [ ] If a feature was added make sure to add it to the corresponding [scripts file](https://github.com/OpenBB-finance/OpenBBTerminal/tree/main/scripts).

# Others
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] My code passes all the checks pylint, flake8, black, ... To speed up development you should run `pre-commit install`.
- [ ] New and existing unit tests pass locally with my changes. You can test this locally using `pytest tests/...`.

### CODE REVIEW

**Comment by DidierRLopes** [comment]
Don't forget to update the website content as well 😄

**Comment by jose-donato** [comment]
> Don't forget to update the website content as well 😄

Yup it's still a WIP

**Comment by DidierRLopes** [comment]
Can we allow to not add a "-e":

Screenshot 2022-07-26 at 01 08 05

File: openbb_terminal/cryptocurrency/due_diligence/ccxt_model.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+    Returns
+    -------
+    List[str]
+        list of all the exchanges supported by ccxt
+    """
+    return ccxt.exchanges
+
+
+def get_orderbook(exchange_id: str, coin: str, vs: str):
```
Comment (DidierRLopes) [inline openbb_terminal/cryptocurrency/due_diligence/ccxt_model.py:23]: Can we add return type with the `-> Dict`

File: openbb_terminal/cryptocurrency/due_diligence/ccxt_view.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    Parameters
+    ----------
+    exchange : str
+        exchange id
+    coin : str
+        coin symbol
+    vs : str
+        currency to compare coin against
+    export : str
+        Export dataframe data to csv,json,xlsx file"""
```
Comment (DidierRLopes) [inline openbb_terminal/cryptocurrency/due_diligence/ccxt_view.py:34]: The `"""` should be 1 line below

File: openbb_terminal/cryptocurrency/due_diligence/ccxt_view.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    exchange : str
+        exchange id
+    coin : str
+        coin symbol
+    vs : str
+        currency to compare coin against
+    top : int
+        number of trades to display
+    export : str
+        Export dataframe data to csv,json,xlsx file"""
```
Comment (DidierRLopes) [inline openbb_terminal/cryptocurrency/due_diligence/ccxt_view.py:67]: `"""` one line below

File: openbb_terminal/cryptocurrency/due_diligence/dd_controller.py [hunk 4]
```diff
@@ -114,11 +115,13 @@ def __init__(
...
         self.messari_timeseries = []
         df_mt = messari_model.get_available_timeseries()
+        self.ccxt_exchanges = ccxt_model.get_exchanges()
+
         if not df_mt.empty:
             self.messari_timeseries = df_mt.index.to_list()
         if session and obbff.USE_PROMPT_TOOLKIT:
             choices: dict = {c: {} for c in self.controller_choices}
-            choices["load"]["--source"] = {c: None for c in CRYPTO_SOURCES.keys()}
+            choices["ob"]["-e"] = {c: None for c in self.ccxt_exchanges}
```
Comment (DidierRLopes) [inline openbb_terminal/cryptocurrency/due_diligence/dd_controller.py:124]: Can we add: 
```
choices["ob"] = {c: None for c in self.ccxt_exchanges}
choices["ob"]["-e"] = {c: None for c in self.ccxt_exchanges}
choices["trades"] = {c: None for c in self.ccxt_exchanges}
choices["trades"]["-e"] = {c: None for c in self.ccxt_exchanges}
```

**Review by DidierRLopes — CHANGES_REQUESTED** [review]
Couple of requests

**Review by DidierRLopes — APPROVED** [review]

**Pull request merged**